src/syscheckd/db/fim_db.c

```diff
@@ -845,13 +853,22 @@ int fim_db_get_checksum_range(fdb_t *fim_sql,
     //Calculate checksum of the second half
     for (i = m; i < n; i++) {
         char *path, *checksum;
-        if (sqlite3_step(fim_sql->stmt[RANGE_QUERY[type]]) != SQLITE_ROW) {
-            merror("Step error getting path range, second half 'start %s' 'top %s' (i:%d): %s", start, top, i,
-                   sqlite3_errmsg(fim_sql->db));
+
+        result = sqlite3_step(fim_sql->stmt[RANGE_QUERY[type]]);
+
+        if (result != SQLITE_ROW) {
+            if (result == SQLITE_DONE) {
+                mdebug2("Received a synchronization message with empty range, first half 'start %s' 'top %s' (i:%d)", start, top, i);
+            } else {
+                merror("Step error getting path range, second half 'start %s' 'top %s' (i:%d): %s", start, top, i,
+                       sqlite3_errmsg(fim_sql->db));
+            }
+
```

Comment:
I know we've talked about this, but I still like the look on this check better if it were a switch-case rather than a nested if, even if we need to repeat the `os_free` statements.
```c
        switch (sqlite3_step(fim_sql->stmt[RANGE_QUERY[type]])) {
        case SQLITE_ROW:
            break;

        case SQLITE_DONE:
            mdebug2("Received a synchronization message with empty range, first half 'start %s' 'top %s' (i:%d)", start,
                    top, i);
            os_free(*str_pathlh);
            os_free(*str_pathuh);
            return FIMDB_ERR;

        default:
            merror("Step error getting path range, first half 'start %s' 'top %s' (i:%d): %s", start, top, i,
                   sqlite3_errmsg(fim_sql->db));
            os_free(*str_pathlh);
            os_free(*str_pathuh);
            return FIMDB_ERR;
        }
```

Same goes for the check on the first half of the algorithm, could we change them?

Reply:
Done in [commit](https://github.com/wazuh/wazuh/pull/8941/commits/2d98a248298d001f88d7f4b5ac067c72f94cb74e)